src/kademlia/node.cpp

```diff
@@ -1191,28 +1190,60 @@ void node::write_nodes_entries(sha1_hash const& info_hash
 		if (wanted_node == m_nodes.end()) continue;
 		std::vector<node_entry> n;
 		wanted_node->second->m_table.find_node(info_hash, n, 0);
-		r[wanted_node->second->protocol_nodes_key()] = write_nodes_entry(n);
+		r[wanted_node->second->protocol().nodes_key()] = write_nodes_entry(n);
 	}
 }
 
-node::protocol_descriptor const& node::map_protocol_to_descriptor(udp protocol)
+protocol_descriptor::protocol_descriptor(udp protocol)
```

Comment:
if you did, a lot of these functions could just not refer to ``m_data``, and maybe even be defined inline above.